### PR TITLE
[FIX] mail: delete messages and activities on module uninstallation

### PR DESCRIPTION
**Pull request by Julien00859**
When uninstalling a module, the messages and the activities linked
to that module are not deleted leading to ghost msg/acti after
reinstalling the module.

This is due to the lack of foreign keys in the definition of the
chatter making it impossible to cascade delete.

opw-1928208

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

**Comment by Julien00859** [comment]
Hello @rco-odoo @Elkasitu

File: addons/mail/models/ir_model.py [hunk 1]
```diff
@@ -16,9 +16,14 @@ class IrModel(models.Model):
...
 
     def unlink(self):
-        # Delete followers for models that will be unlinked.
+        # Delete followers, messages and activities for models that will be unlinked.
+        models = tuple(self.mapped('model'))
         query = "DELETE FROM mail_followers WHERE res_model IN %s"
-        self.env.cr.execute(query, [tuple(self.mapped('model'))])
+        self.env.cr.execute(query, [models])
+
+        self.env['mail.activity'].search([('res_model', 'in', models),]).unlink()
```
Comment (Elkasitu) [inline addons/mail/models/ir_model.py]: is this necessary? cf. https://github.com/odoo/odoo/blob/12.0/addons/mail/models/mail_activity.py#L112

Comment (Julien00859) [reply addons/mail/models/ir_model.py]: letme verify...

Comment (Julien00859) [reply addons/mail/models/ir_model.py]: yup not required

**Comment by Julien00859** [comment]
@Xavier-Do

**Comment by Xavier-Do** [comment]
LGTM

**Comment by Yenthe666** [comment]
@robodoo don't block the fix for my bug report you silly bot! I needs this.

**Comment by nim-odoo** [comment]
robodoo r+

**Comment by simahawk** [comment]
@nim-odoo what about adding tests? So next time this does not happen anymore? :pray:

**Comment by Julien00859** [comment]
@simahawk nim is not responsible to write tests on the code I write. What kind of test are you talking about ? Testing that messages are deleted on a module uninstallation ?

**Comment by simahawk** [comment]
> What kind of test are you talking about ? Testing that messages are deleted on a module uninstallation ?

what else? :)
Regarding responsible: up to you to do them up to reviewers to ask for them ;)

**Pull request closed without merging**